### PR TITLE
Initial update of the README file

### PR DESCRIPTION
**Pull request by yanjos-dev**
This is a first pass on updating the README file. There will need to be a follow up for some of the links.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,78 @@
...
+
+This library is distributed under the [MIT Open Source License](LICENSE).
+
+This library has gone through code quality checks including verification that no function has a [GNU Complexity](https://www.gnu.org/software/complexity/manual/complexity.html) score over 8. This library has also undergone static code analysis from [Coverity static analysis](https://scan.coverity.com/).
+
+See memory requirements for this library here. (TODO: Add a link to the memory req doxygen)
```
Comment (sarenameas) [inline README.md:9]:
```suggestion
See memory requirements for this library [here](https://docs.aws.amazon.com/embedded-csdk/202012.00/lib-ref/libraries/aws/ota-for-aws-iot-embedded-sdk/docs/doxygen/output/html/index.html#ota_memory_requirements).
```
Place holder this is the structure of the link

File: README.md [hunk 2]
```diff
@@ -1,14 +1,78 @@
-# OTA For AWS IoT Embedded SDK
+# AWS IoT Over-the-Air (OTA) Updates
 
-## Porting
+The OTA library enables you to manage the notification of a newly available update, download the update, and perform cryptographic verification of the firmware update. Using the library, you can logically separate firmware updates from the application running on your devices. The OTA library can share a network connection with the application, saving memory in resource-constrained devices. In addition, the OTA library lets you define application-specific logic for testing, committing, or rolling back a firmware update. The library supports different application protocols like Message Queuing Telemetry Transport (MQTT) and Hypertext Transfer Protocol (HTTP) and provides various configuration options you can fine tune depending on network type and conditions.
```
Comment (sarenameas) [inline README.md:3]:
```suggestion
The OTA library enables you to manage the notification of a newly available update, download the update, and perform cryptographic verification of the firmware update. Using the library, you can logically separate firmware updates from the application running on your devices. The OTA library can share a network connection with the application, saving memory in resource-constrained devices. In addition, the OTA library lets you define application-specific logic for testing, committing, or rolling back a firmware update. The library supports different application protocols like Message Queuing Telemetry Transport (MQTT) and Hypertext Transfer Protocol (HTTP), and provides various configuration options you can fine tune depending on network type and conditions.
```

File: README.md [hunk 3]
```diff
@@ -1,14 +1,78 @@
...
+For a CMake example of building the AWS IoT Over the Air (OTA) Updates library with the `otaFilePaths.cmake` file, refer to the `coverity_analysis` library target in the [test/CMakeLists.txt](test/CMakeLists.txt) file.
+
+## Building Unit Tests
+### Checkout CMock Submodule
+By default, the submodules in this repository are configured with `update=none` in [.gitmodules](.gitmodules) to avoid increasing clone time and disk space usage of other repositories (like [AWS IoT Device SDK for Embedded C](https://github.com/aws/aws-iot-device-sdk-embedded-C) that submodules this repository).
+
+
+To build unit tests, the submodule dependency of CMock is required. Use the following command to clone the submodule:
+```
+git submodule update --checkout --init --recursive --test/unit-test/CMock
```
Comment (sarenameas) [inline README.md:31]:
```suggestion
git submodule update --checkout --init --recursive test/unit-test/CMock
```

File: README.md [hunk 4]
```diff
@@ -1,14 +1,78 @@
...
+
+1. Run this command to build the library and unit tests: `make -C build all`
+
+1. The generated test executables will be present in `build/bin/tests` folder.
+
+1. Run `cd build && ctest` to execute all tests and view the test run summary.
+
+## Reference examples
+
+Please refer to the demos of the AWS IoT Device Shadow library in the following locations for reference examples on POSIX and FreeRTOS platforms:
```
Comment (sarenameas) [inline README.md:54]: Shadow

**Review by nihaljamazon — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1,14 +1,78 @@
-# OTA For AWS IoT Embedded SDK
+# AWS IoT Over-the-Air (OTA) Updates
```
Comment (tanmoysen) [inline README.md:1]: AWS IoT Over-the-air Updates

**Review by sarenameas — APPROVED** [review]

**Review by gkwicker — APPROVED** [review]

**Pull request merged**